### ISSUE TITLE
Building contiv network container fails

### ISSUE THREAD

**Issue by gerritbinnenmars** [contiv/netplugin#779]
After git clone contiv/netplugin the following command fails:
[root@localhost netplugin]# scripts/netContain/BuildContainer.sh 
+ ARG1=none
+ '[' none == reinit ']'
+ sudo modprobe openvswitch
+ imgName=contiv-network-plugin
+ dockerBuildIt . contiv-network-plugin
++ docker build .
++ grep 'Successfully built'
++ cut -d ' ' -f 3
The command '/bin/sh -c make build' returned a non-zero code: 2
+ imgId=
+ [[ '' =~ [0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f] ]]
+ echo 'contiv-network-plugin Image was not built properly'
contiv-network-plugin Image was not built properly
+ return 255
+ '[' 255 '!=' 0 ']'
+ echo 'Failed building Contiv Image Bailing out Err 0'
Failed building Contiv Image Bailing out Err 0
+ exit

**Comment by gerritbinnenmars** [comment]
Please, upgrade the Dockerfile to: FROM golang:1.7 to solve this issue

**Comment by dseevr** [comment]
The step the build process is failing on doesn't have anything to do with Go:

```
Step 8/14 : COPY ./netplugin /contiv/bin/
lstat netplugin: no such file or directory
```

@unclejack do you know what's going on here?

**Comment by unclejack** [comment]
@dseevr: Have you run `make run-build` first?

**Comment by dseevr** [comment]
@unclejack I ran `./scripts/netContain/BuildContainer.sh` as documented [here](https://github.com/contiv/netplugin/tree/master/scripts/netContain#introduction) and this is the step the `docker build` command failed on

**Comment by gerritbinnenmars** [comment]
Correct, the original error occurs in make run-build in the go install step in the subdirectory netplugin.
Below it the output when this step is done manually. See also: https://github.com/docker/engine-api/issues/385

root@5ddc7f2af0b2:/go/src/github.com/contiv/netplugin# GOGC=1500 go install -v ./netplugin
github.com/contiv/netplugin/vendor/github.com/docker/engine-api/client/transport
github.com/contiv/netplugin/version
# github.com/contiv/netplugin/vendor/github.com/docker/engine-api/client/transport
vendor/github.com/docker/engine-api/client/transport/tlsconfig_clone.go:10: c.Clone undefined (type *tls.Config has no field or method Clone)

**Comment by unclejack** [comment]
@gerritbinnenmars: I'll send a PR shortly to fix this issue. I'm sorry you've had to deal with this problem.

**Comment by rootsongjc** [comment]
### make build error
**Error**
```
# github.com/contiv/ofnet
../ofnet/ofnetBgp.go:431: not enough arguments in call to server.WatchBestPath
../ofnet/ofnetBgp.go:609: not enough arguments in call to self.bgpServer.GetNeighbor
github.com/contiv/netplugin/utils/netutils
```
**Detail**
```
[root@node1 netplugin]# NET_CONTAINER_BUILD=1 make build
make start
make[1]: Entering directory `/root/github/netplugin'
make[1]: Nothing to be done for `start'.
make[1]: Leaving directory `/root/github/netplugin'
make ssh-build
make[1]: Entering directory `/root/github/netplugin'
cd /go/src/github.com/contiv/netplugin && make run-build
make[2]: Entering directory `/root/go/src/github.com/contiv/netplugin'
./scripts/deps
+++ check go version
+++ gofmt core drivers mgmtfn netctl netmaster netplugin state utils version
+++ golint core drivers mgmtfn netctl netmaster netplugin state utils version
...
github.com/contiv/ofnet
# github.com/contiv/ofnet
../ofnet/ofnetBgp.go:431: not enough arguments in call to server.WatchBestPath
../ofnet/ofnetBgp.go:609: not enough arguments in call to self.bgpServer.GetNeighbor
github.com/contiv/netplugin/utils/netutils
...
make[2]: *** [run-build] Error 2
make[2]: Leaving directory `/root/go/src/github.com/contiv/netplugin'
make[1]: *** [ssh-build] Error 2
make[1]: Leaving directory `/root/github/netplugin'
make: *** [build] Error 2
```
**Environment**
CentOS7.2.1511
go version go1.7rc6 linux/amd64

**Comment by jojimt** [comment]
I am wondering how this breakage slipped through. Does this mean that this build is not part of our sanity? If so, we need to fix that as well.

**Comment by unclejack** [comment]
PR #785 addresses the encountered issues.

I'll submit a PR after #785 is merged to add it to the tests.

**Comment by unclejack** [comment]
This was fixed by #780. That Dockerfile was the one which was causing this particular error.

#785 cleans up the image building script to make it more reliable and to avoid otther issues. #785 also makes the script more robust to avoid issues in a future Docker release.
